jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/lock/CustomRuntimeManagerLockStrategyTest.java

```diff
@@ -89,4 +92,51 @@ public void testInterruptible() throws Exception {
         Assert.assertTrue(executorService.isTerminated());
         logger.info("exiting test TimeoutRuntimeManagerLockStrategy");
     }
+
+
+    @Test(timeout = 10000)
+    public void testWatcher() throws Exception {
+        RuntimeManagerLockWatcherSingletonService watcher = RuntimeManagerLockWatcherSingletonService.reference();
+        watcher.setMaxLockProcessingTime(200L);
+        watcher.setWatchLockPolling(0L);
+        watcher.start();
+
+        SelfReleaseRuntimeManagerLock lockFirst = new SelfReleaseRuntimeManagerLock();
+        lockFirst.lock();
+        watcher.watch(1, lockFirst);
+        Thread.sleep(100L);
+        SelfReleaseRuntimeManagerLock lockSecond = new SelfReleaseRuntimeManagerLock();
+        lockSecond.unlock();
+        watcher.watch(2, lockSecond);
+
+        long wait = watcher.tryForcedUnlock();
+        Assert.assertTrue(wait > 0);
+
+        Thread.sleep(wait);
+        wait = watcher.tryForcedUnlock();
+        Assert.assertTrue(wait > 0);
```

Comment:
Here we could check something like:

```suggestion
        Assert.assertTrue(wait > 0);
        Assert.assertFalse(watcher.isWatched(1));
        Assert.assertFalse(lockFirst.isHeldByCurrentThread());
        Assert.assertTrue(watcher.isWatched(2));
        Assert.assertFalse(lockSecond.isHeldByCurrentThread());
```
Though not sure about that unlocked second lock.

Reply:
done